Source/d_iwad.c

```diff
@@ -18,12 +18,32 @@
 
 #include <stdlib.h>
 
-#include "doomtype.h"
+#include "i_system.h"
 #include "d_io.h"
 #include "d_iwad.h"
 #include "m_argv.h"
 #include "m_misc2.h"
 
+static const iwad_t iwads[] =
+{
+    { "doom2.wad",     doom2,      commercial, "Doom II" },
+    { "plutonia.wad",  pack_plut,  commercial, "Final Doom: Plutonia Experiment" },
+    { "tnt.wad",       pack_tnt,   commercial, "Final Doom: TNT: Evilution" },
+    { "doom.wad",      doom,       retail,     "Doom" },
```

Comment:
Do we have any chance to detect non-retail, i.e. non-Ultimate Doom 1?

Reply:
It seems not. So we still need to check the contents of the IWAD if `filename == "doom.wad"`.